compiler/src/test/scala/org/scalajs/core/compiler/test/JSInteropTest.scala

```diff
@@ -1002,6 +1002,76 @@ class JSInteropTest extends DirectTest with TestHelpers {
   }
 
   @Test
+  def warnPrivateMemberInNative: Unit = {
+
+    """
+    @js.native
+    @JSGlobal
+    class A extends js.Object {
+      private[this] val a: Int = js.native
+      private val b: Int = js.native
+      private[A] val c: Int = js.native
```

Comment:
Could you add a var as well, just to make sure we have the right number of warnings?